server/gpgkey/gpgkey.go

```diff
@@ -81,6 +82,11 @@ func (s *Server) CreateGnuPGPublicKey(ctx context.Context, q *gpgkeypkg.GnuPGPub
 		return nil, err
 	}
 
+	keyData := strings.TrimSpace(q.Publickey.KeyData)
+	if keyData == "" || !strings.HasPrefix(keyData, "-----BEGIN PGP PUBLIC KEY BLOCK-----") || !strings.HasSuffix(keyData, "-----END PGP PUBLIC KEY BLOCK-----") {
```

Comment:
This now disallows non ascii armoured keys.

Reply:
Yes, this was a deliberate decision. Do you think people would require the binary format to be supported when importing their public keys to ArgoCD? I was always under the assumption that the ASCII armoured format would be good enough for most if not any tasks, but that's just me. If there are good reasons to support binary key format, I can move the checks elsewhere - this is mainly a convenience check for the web UI users who are going to paste their AA key to the text field upon adding their key.

Reply:
I've always just done non-ascii armoured keys, because k8s config maps base64 them anyway, and I've always hated double base 64 encoded data,

Reply:
Thanks for the feedback. Let me check how we can best support both, with good feedback to the user about the validity of their input without disabling import of binary format keys.